controllers/numaresourcesoperator_controller_test.go

```diff
@@ -446,7 +448,136 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 
 			Expect(len(nroUpdated.Status.MachineConfigPools)).To(Equal(1))
 			Expect(nroUpdated.Status.MachineConfigPools[0].Name).To(Equal(mcp.Name))
-			// TODO check the actual returned config and the daemonset in status, we need to force the NRO condition as Available for this.
+
+			conf := nropv1.DefaultNodeGroupConfig()
+			Expect(nroUpdated.Status.NodeGroups[0].Config).To(Equal(conf), "node group config was not updated in the operator status")
+		})
+
+		When("multiple node groups are configured with different config", func() {
+			It("should create all RTE daemonsets and report the status respectively per node group", func(ctx context.Context) {
+				mcp1Name := "test1"
+				mcp2Name := "test2"
+				label1 := map[string]string{
+					"test1": "test1",
+				}
+				label2 := map[string]string{
+					"test2": "test2",
+				}
+
+				defaultConf := nropv1.DefaultNodeGroupConfig()
+
+				conf1 := defaultConf.DeepCopy()
+				pfpModeDisabled := nropv1.PodsFingerprintingDisabled
+				rteMode := nropv1.InfoRefreshPauseEnabled
+				conf1.PodsFingerprinting = &pfpModeDisabled
+				conf1.InfoRefreshPause = &rteMode
+
+				conf2 := defaultConf.DeepCopy()
+				pfpModeEnabled := nropv1.PodsFingerprintingEnabled
+				refMode := nropv1.InfoRefreshPeriodicAndEvents
+				conf2.PodsFingerprinting = &pfpModeEnabled
+				conf2.InfoRefreshMode = &refMode
+				conf2.Tolerations = []corev1.Toleration{
+					{
+						Key:    "foo",
+						Value:  "1",
+						Effect: corev1.TaintEffectNoSchedule,
+					},
+				}
+
+				ng1 := nropv1.NodeGroup{
+					PoolName: &mcp1Name,
+					Config:   conf1,
+				}
+				ng2 := nropv1.NodeGroup{
+					PoolName: &mcp2Name,
+					Config:   conf2,
+				}
+
+				mcp1 := testobjs.NewMachineConfigPool(mcp1Name, label1, &metav1.LabelSelector{MatchLabels: label1}, &metav1.LabelSelector{MatchLabels: label1})
+				mcp2 := testobjs.NewMachineConfigPool(mcp2Name, label2, &metav1.LabelSelector{MatchLabels: label2}, &metav1.LabelSelector{MatchLabels: label2})
```

Comment:
nit: variable definition can be as close as possible to the first usage to minimize scope (and thus cognitive load)

Reply:
please do NOT resubmit only to fix this